### ISSUE TITLE
Force-terminate operation for stuck orchestrations

### ISSUE THREAD

**Issue by cgillum** [Azure/azure-functions-durable-extension#2081]
### Is your feature request related to a problem? Please describe.
The current terminate implementation requires orchestrations to be loaded in memory before the terminate operation can be processed. This is a problem for orchestrations that are stuck because their histories can't be loaded (perhaps due to memory restrictions). This also prevents the re-use of the orchestration ID since orchestrations can only be recreated when they're in a Completed, Failed, or Terminated state.

### Are there any existing GitHub discussions or issues filed that help give some context to this proposal?
There probably are several, but I haven't gone through all issues to find them. It's an issue that comes up in a lot of different contexts, however.

### Describe the solution you'd like
I think the feature could be as simple as a new `force` flag on the terminate API. When set to `true`, we force the terminating the instance in a way that doesn't require it to be reloaded (this may be challenging to do in the Azure Storage provider without a noticeable performance impact, FYI).

### Describe alternatives you've considered
Another option would be a "force purge" API, which just purges the instance completely, regardless of what state it's in. It's a bit more dramatic in that you lose the metadata, but it's likely going to be cheaper to implement.

**Comment by davidmrdavid** [comment]
I agree this would be great to have, so thank you for starting this discussion!

A related design question here is: how do we expect users to know that their orchestration's history cannot be loaded? After all, they need to know that to determine that the _force_ flag is required. Do they already have a beginner-friendly way of diagnosis this issue? Otherwise, I think we should also consider giving them that diagnosis tool.

**Comment by cgillum** [comment]
There isn't a simple way to know what exactly is preventing a terminate from completing. I used the history load failure as an example, but there are other possibilities as well. The idea is that this can be a tool that can be used if a normal usage of Terminate isn't working for any reason.

**Comment by davidmrdavid** [comment]
That makes sense. Just to be sure though: even if we can't diagnose why `Terminate` isn't completing, would we be able to give users a notification that `Terminate` is (or appears to be) failing to succeed?

In other words, how does a user come to the realization that they should _force_ the termination of their app? Is it just observing that the runtime state hasn't changed in some time, or can we give them a tool to make that determination?

**Comment by dashesy** [comment]
Would be nice to able able to purge from Azure portal. There are some instances that failed but show as running and purging them makes sense.

**Comment by YunchuWang** [comment]
Thanks for raising this, @cgillum. Durable Task Scheduler (DTS) now supports force termination through the dashboard's **Force terminate** option; the emergency termination API was introduced in [microsoft/durabletask-protobuf#48](https://github.com/microsoft/durabletask-protobuf/pull/48) and changes instance status directly in storage rather than waiting for a worker to process a termination event. The Azure Storage termination path is still message-based and depends on loading the orchestration history.

Do we still plan to add equivalent support for Azure Storage and other backends, keeping this issue open to track that work, or can this be closed now that DTS supports it?
